### PR TITLE
Allow external ScrollController and FIX inconsistent speed behavior

### PR DESCRIPTION
**Pull request by eltonmorais**
To give the dev more freedom to control the scroll screen I have allowed the use of a external ScrollController. It's not mandatory. If we get a null on the parameter, we use the same method as before.

I found a inconsistent behavior on the speed scroll. If I start the scroll from the top, a speed value give me a specific speed. But if I start the scroll in the midle or end of the music, then the same speed value give me a different speed.

Now same speed value allways give the same speed scrolling. What allow to create a consistent speed controller.

### CODE REVIEW

**Review by paurakhsharma — APPROVED** [review]
Looks good. Thank you so much!

**Comment by paurakhsharma** [comment]
@eltonmorais Can you please fix the formatting issue and update the PR?

**Review by paurakhsharma — CHANGES_REQUESTED** [review]
I don't think we need to and bottom padding. That should be achievable using leading and training widgets.

File: lib/src/lyrics_renderer.dart [hunk 1]
```diff
@@ -8,7 +8,10 @@ class LyricsRenderer extends StatefulWidget {
   final TextStyle textStyle;
   final TextStyle chordStyle;
   final bool showChord;
-  final Function onTapChord;
+  final Function(String) onTapChord;
+  final ScrollController? scrollController;
+  final double topPadding;
+  final double bottomPadding;
```
[the remark this answers is not in the document]
Comment (eltonmorais) [reply lib/src/lyrics_renderer.dart:14]: Agree.

[the remark this answers is not in the document]
Comment (paurakhsharma) [reply lib/src/lyrics_renderer.dart:14]: Can you please remove them and push the changes? I will merge it after that.

**Review by paurakhsharma — APPROVED** [review]

**Review by paurakhsharma — CHANGES_REQUESTED** [review]
Please remove topPadding and bottomPadding. It can be handeled by leading and trailling widget.

**Comment by paurakhsharma** [comment]
Closing because the original author didn't update the PR. Similar functionality implemented by #31

**Pull request closed without merging**